src/main/java/io/github/perrymant/moneymaker/FileIO.java

```diff
@@ -0,0 +1,31 @@
+package io.github.perrymant.moneymaker;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+class FileIO {
+
+    static String read(String fileName) throws IOException {
+        BufferedReader text = new BufferedReader(new FileReader(fileName));
+        List<String> data = new ArrayList<>();
+        String s;
+        while ((s = text.readLine()) != null) {
+            data.add(s + "\n");
+        }
+        text.close();
+        return String.join("", data);
+    }
+
+    void write(String data, String fileName) throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(fileName));
+        writer.write(data);
+        writer.close();
```

Comment:
Have a look at what try-with-resources or `finally` does in Java. Essentially we need to guarantee that `writer.close()` gets called, and if `writer.write(data)` throws an exception we end up not closing it.